test/update_test.go

```diff
@@ -0,0 +1,145 @@
+				Then(clientv3.OpPut("updateNewKey", "testValue")).
+				Else(clientv3.OpGet("updateNewKey", clientv3.WithRange(""))).
+				Commit()
+
+			g.Expect(err).To(BeNil())
+			g.Expect(resp.Succeeded).To(BeTrue())
+		}
+
+		{
+			resp, err := client.Get(ctx, "updateNewKey", clientv3.WithRange(""))
+			g.Expect(err).To(BeNil())
+			g.Expect(resp.Kvs).To(HaveLen(1))
+			g.Expect(resp.Kvs[0].Key).To(Equal([]byte("updateNewKey")))
+			g.Expect(resp.Kvs[0].Value).To(Equal([]byte("testValue")))
+			g.Expect(resp.Kvs[0].ModRevision).To(BeNumerically("==", resp.Kvs[0].CreateRevision))
+		}
+	})
+
+	t.Run("UpdateExisting", func(t *testing.T) {
+		g := NewWithT(t)
+
+		var lastModRev int64
+
+		{
+			resp, err := client.Txn(ctx).
+				If(clientv3.Compare(clientv3.ModRevision("updateExistingKey"), "=", 0)).
+				Then(clientv3.OpPut("updateExistingKey", "testValue1")).
+				Commit()
+
+			g.Expect(err).To(BeNil())
+			g.Expect(resp.Succeeded).To(BeTrue())
+			lastModRev = resp.Responses[0].GetResponsePut().Header.Revision
+		}
+
+		{
+			resp, err := client.Txn(ctx).
+				If(clientv3.Compare(clientv3.ModRevision("updateExistingKey"), "=", lastModRev)).
+				Then(clientv3.OpPut("updateExistingKey", "testValue2")).
+				Else(clientv3.OpGet("updateExistingKey", clientv3.WithRange(""))).
+				Commit()
+
+			g.Expect(err).To(BeNil())
+			g.Expect(resp.Succeeded).To(BeTrue())
+		}
+
+		{
+			resp, err := client.Get(ctx, "updateExistingKey", clientv3.WithRange(""))
+			g.Expect(err).To(BeNil())
+			g.Expect(resp.Kvs).To(HaveLen(1))
+			g.Expect(resp.Kvs[0].Key).To(Equal([]byte("updateExistingKey")))
+			g.Expect(resp.Kvs[0].Value).To(Equal([]byte("testValue2")))
+			g.Expect(resp.Kvs[0].ModRevision).To(BeNumerically(">", resp.Kvs[0].CreateRevision))
+		}
+	})
+
+	// Trying to update an old revision(in compare) should fail
+	t.Run("UpdateOldRevisionFails", func(t *testing.T) {
+		g := NewWithT(t)
+
+		var lastModRev int64
+
+		{
+			resp, err := client.Txn(ctx).
+				If(clientv3.Compare(clientv3.ModRevision("updateOldRevKey"), "=", 0)).
+				Then(clientv3.OpPut("updateOldRevKey", "testValue1")).
+				Commit()
+
+			g.Expect(err).To(BeNil())
+			g.Expect(resp.Succeeded).To(BeTrue())
+			lastModRev = resp.Responses[0].GetResponsePut().Header.Revision
+		}
+
+		{
+			resp, err := client.Txn(ctx).
+				If(clientv3.Compare(clientv3.ModRevision("updateOldRevKey"), "=", lastModRev)).
+				Then(clientv3.OpPut("updateOldRevKey", "testValue2")).
+				Else(clientv3.OpGet("updateOldRevKey", clientv3.WithRange(""))).
+				Commit()
+
+			g.Expect(err).To(BeNil())
+			g.Expect(resp.Succeeded).To(BeTrue())
+		}
+
+		{
+			resp, err := client.Txn(ctx).
+				If(clientv3.Compare(clientv3.ModRevision("updateOldRevKey"), "=", lastModRev)).
+				Then(clientv3.OpPut("updateOldRevKey", "testValue2")).
+				Else(clientv3.OpGet("updateOldRevKey", clientv3.WithRange(""))).
+				Commit()
+
+			g.Expect(err).To(BeNil())
+			g.Expect(resp.Succeeded).To(BeFalse())
```

Comment:
What do the responses look like in this situation? Is it an empty list, or a list of size 1 with more information about the failure? Let's add validations for this.